src/hoc/BadgerBase/BadgerBase.js

```diff
@@ -368,7 +505,7 @@ const BadgerBase = (Wrapped: React.AbstractComponent<any>) => {
 					this.startRepeatable();
 				}
 
-				if (tokenId !== prevTokenId) {
+				if (tokenId !== prevTokenId && !paymentRequestUrl) {
```

Comment:
Can we do all the `setupCoinMeta` in the same way as before, and just set the relevant fields in the `paymentRequest` setup step?
This would allow the `setupCoinMeta` to stay simple, and handle the payment request added complexity locally in the payment request flows.

Reply:
The code must over-ride `props` if the invoice SLP type does not match the prop type. This could be handled by setting `tokenId` in both `props` and `state`; exporting the `state` version, and having `BadgerButton` and `BadgerBadge` only read these and not the `props` version. I think this would be more complicated than what we have right now. Right now, program logic is the same for cases with no `paymentRequestUrl` parameter. If `paymentRequestUrl` is set, then information from the invoice over-rides anything in props (should be this way to prevent anyone from using `badger-react-components` to display incorrect information about an invoice by entering misleading props).

Let me know thoughts on improving this. I don't think the current implementation is perfect, but I wanted to make sure that in the case of any invoice being displayed, the display information can come only from that invoice.